app/classifier/feedback/feedback-ruleset.js

```diff
@@ -0,0 +1,61 @@
+export default class FeedbackRuleSet {
+  constructor(subject, task) {
+    this._subject = subject;
+    this._task = task;
+    const activeTypes = this._getActiveTypes();
+    this.rules = this._generateRules(activeTypes);
+  }
+
+  _generateRules(activeTypes) {
+    return Object.keys(activeTypes).reduce((ruleset, key) => {
+      const defaults = this._getDefaultsForType(activeTypes[key]);
+      const subj = this._getMetadataFieldsForType(key);
+      const newRule = Object.assign({}, defaults, subj, { _defaults: defaults });
+      return ruleset.concat(newRule);
+    }, []);
+  }
+
+  _getActiveTypes() {
+    // Find the feedback types required in the subject metadata and check for valid active rules in the task definition.
+    const { metadata } = this._subject;
+    const { feedback } = this._task;
+    return Object.keys(metadata).reduce((types, key) => {
+      const match = key.match(/#feedback_(\d*?)_type/);
+      if (match) {
+        const index = match[1];
+        const type = metadata[key];
+        const typeInTaskDefinition = feedback.types.find((feedbackItem) => feedbackItem.id === type);
```

Comment:
`Array.find` isn't supported in IE, so this will need a polyfill. Polyfill.io can polyfill it, just needs to be added to the param list in the source url.